src/TeslaBMS.cpp

```diff
@@ -67,133 +62,106 @@ void module_array_maker(ModuleList *list){
 
 
 /* Module encode callback */
-bool modules_encode(pb_ostream_t *stream, const pb_field_iter_t *field, void * const *arg)
-{    
+bool modules_encode(pb_ostream_t *stream, const pb_field_iter_t *field, void * const *arg) {
     ModuleList *source = (ModuleList*)(*arg);
 
     for(int i=0; i<bms.getNumOfModules();i++)
     {
-        printf(" \n Mod %i is at %f \n",(int)source->modarr[i].id, source->modarr[i].moduleVoltage);
+        printf("\nEncoding TeslaPack_Module[%i].moduleVoltage %f \n",(int)source->modarr[i].id, source->modarr[i].moduleVoltage);
         if (!pb_encode_tag_for_field(stream, field))
         {
             const char * error = PB_GET_ERROR(stream);
             printf("encode_modules error: %s", error);
             return false;
         }
-        if (stream, TeslaBMS_Pack_Module_fields, &source->modarr[i])
-        {
-            printf("\n %i \n",stream-> bytes_written);
+        
+        bool status = pb_encode_submessage(stream, TeslaBMS_Pack_Module_fields, &source->modarr[i]);

```

Comment:
crux of the bug is this line here

Reply:
🤦‍♂️ughhhh good catch, looks like I almost had it :)